libraries/core_libs/network/src/tarcap/shared_states/peers_state.cpp

```diff
@@ -28,18 +28,32 @@ std::shared_ptr<TaraxaPeer> PeersState::getPendingPeer(const dev::p2p::NodeID& n
   return nullptr;
 }
 
-std::pair<std::shared_ptr<TaraxaPeer>, bool> PeersState::getAnyPeer(const dev::p2p::NodeID& node_id) const {
+std::pair<std::shared_ptr<TaraxaPeer>, std::string> PeersState::getPacketSenderPeer(
+    const dev::p2p::NodeID& node_id, SubprotocolPacketType packet_type) const {
   std::shared_lock lock(peers_mutex_);
 
+  // If peer is in peers_, it means he already sent us initial status packet and
+  // we can receive/send any kind of packet from/to him
   if (const auto it_peer = peers_.find(node_id); it_peer != peers_.end()) {
-    return {it_peer->second, false};
+    return {it_peer->second, ""};
   }
 
+  // If peer is in pending_peers_, it means he has not yet sent us initial status packet and
+  // we can receive/send only StatusPacket from/to him
   if (const auto it_peer = pending_peers_.find(node_id); it_peer != pending_peers_.end()) {
-    return {it_peer->second, true};
+    if (packet_type == SubprotocolPacketType::StatusPacket) {
+      return {it_peer->second, ""};
+    } else {
```

Comment:
hmm if we dont search pending pears then we wont be able to differentiate between peer not being in peers at all vs peer being just in pending peers - might be useful to know this info and most of the time we wont get here because function will return true after peer is found in peers_